src/Tribe/Repositories/Post_Repository.php

```diff
@@ -0,0 +1,245 @@
+							$the_order                   = is_numeric( $k ) ? $default_order : $v;
+							$normalized[ $the_order_by ] = $the_order;
+						}
+						$this->query_args['orderby'] = $normalized;
+						$this->query_args['orderby'] = array_merge( $this->query_args['orderby'], $add );
+					}
+					break;
+			}
+		}
+	}
+
+	/**
+	 * Overrides the base method to correctly handle the `order_by` clauses before.
+	 *
+	 * The Event repository handles ordering with some non trivial logic and some query filtering.
+	 * To avoid the "stacking" of `orderby` clauses and filters the query filters are added at the very last moment,
+	 * right before building the query.
+	 *
+	 * @since 4.9.7
+	 *
+	 * @return \WP_Query The built query object.
+	 */
+	protected function build_query_internally() {
+		$order_by = Arr::get_in_any( [ $this->query_args, $this->default_args ], 'orderby', 'event_date' );
+
+		unset( $this->query_args['orderby'], $this->default_args['order_by'] );
+
+		$this->handle_order_by( $order_by );
+
+		return parent::build_query_internally();
+	}
+
+	/**
+	 * Applies start-date-based ordering to the query.
+	 *
+	 * @since 4.9.7
+	 * @since 4.9.11 Added the `$after` parameter.
+	 *
+	 * @param bool $use_utc      Whether to use the events UTC start dates or their localized dates.
+	 * @param bool $after        Whether to append the order by clause to the ones managed by WordPress or not.
+	 *                           Defaults to `false`,to prepend them to the ones managed by WordPress.
+	 */
+	protected function order_by_date( $use_utc, $after = false ) {
+		global $wpdb;
+
+		$meta_alias = 'event_date';
+		$meta_key   = '_EventStartDate';
+
+		/**
+		 * When the "Use site timezone everywhere" option is checked in events settings,
+		 * the UTC time for event start and end times will be used. This filter allows the
+		 * disabling of that in certain contexts, so that local (not UTC) event times are used.
+		 *
+		 * @since 4.6.10
+		 *
+		 * @param boolean $force_local_tz Whether to force the local TZ.
+		 */
+		$force_local_tz = apply_filters( 'tribe_events_query_force_local_tz', false );
+
+		/*
+		 * The setting is not being forced by means of a call to the `use_utc` method.
+		 * First we check if we've got a UTC ordering request in the `orderby` clause.
+		 * After that if the use of the local (to the event) timezone is being forced by a filter.
+		 * Finally if the timezone setting is set to use the site-wide timezone or not.
+		 */
+		if ( $use_utc || ( ! $force_local_tz && Timezones::is_mode( 'site' ) ) ) {
+			$meta_alias = 'event_date_utc';
+			$meta_key   = '_EventStartDateUTC';
+		}
+
+		$postmeta_table = "orderby_{$meta_alias}_meta";
+
+		$filter_id = 'order_by_date';
+
+		$this->filter_query->join( $wpdb->prepare( "
+			LEFT JOIN `{$wpdb->postmeta}` AS `{$postmeta_table}`
+				ON (
+					`{$postmeta_table}`.post_id = `{$wpdb->posts}`.ID
+					AND `{$postmeta_table}`.meta_key = %s
+				)
+			", $meta_key ), $filter_id, true );
+
```

Comment:
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 15d4f640a445763280e3326885d49b16</sub>